pkg/ddl/rollingback.go

```diff
@@ -760,3 +762,25 @@ func rollingBackAlterConstraint(jobCtx *jobContext, job *model.Job) (ver int64,
 	ver, err = updateVersionAndTableInfoWithCheck(jobCtx, job, tblInfo, true)
 	return ver, errors.Trace(err)
 }
+
+func rollbackModifySchemaReadOnly(jobCtx *jobContext, job *model.Job) (ver int64, err error) {
+	dbInfo, err := jobCtx.metaMut.GetDatabase(job.SchemaID)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+	args, err := model.GetModifySchemaArgs(job)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(err)
+	}
+	dbInfo.ReadOnly = !args.ReadOnly
+	if err = jobCtx.metaMut.UpdateDatabase(dbInfo); err != nil {
+		job.State = model.JobStateCancelled
```

Comment:
is this necessary, seems not all error will set the state, such as `rollingbackAddColumn`